### PR TITLE
Optimization attempt

### PR DESCRIPTION
**Pull request by sabbour**
- Rewrote captainkube
- Updated SignalR NuGet version
- Added clearing functionality
- Renamed dotnetapp chart to parrot

### CODE REVIEW

File: parrot/Hubs/Daemon.cs [hunk 1]
```diff
@@ -36,30 +36,40 @@ public void AddPod(Pod pod)
...
         public void RemovePod(Pod pod)
         {
-            Pods.Remove(Pods.First(x => x.Container == pod.Container));
+            Pods.Remove(Pods.First(x => x.Name == pod.Name));
             DeletedPods.Add(pod.Name);
         }
 
         public void UpdatePod(Pod pod)
         {
-            Pods.First(x => x.Container == pod.Container).Name = pod.Name;
```
Comment (bradygmsft) [inline parrot/Hubs/Daemon.cs]: DAMMIT i'm an idiot. THANK YOU. this was driving me bonkers. #facepalm

File: parrot/Hubs/Daemon.cs [hunk 2]
```diff
@@ -36,30 +36,40 @@ public void AddPod(Pod pod)
...
-            Pods.First(x => x.Container == pod.Container).NameSpace = pod.NameSpace;
-            Pods.First(x => x.Container == pod.Container).Status = pod.Status;
+            Pods.First(x => x.Name == pod.Name).Name = pod.Name;
+            Pods.First(x => x.Name == pod.Name).Container = pod.Container;
+            Pods.First(x => x.Name == pod.Name).NameSpace = pod.NameSpace;
+            Pods.First(x => x.Name == pod.Name).Status = pod.Status;
```
Comment (bradygmsft) [inline parrot/Hubs/Daemon.cs]: i still can't believe i did this. jeez

File: parrot/Hubs/Daemon.cs [hunk 3]
```diff
@@ -36,30 +36,40 @@ public void AddPod(Pod pod)
...
         public void updateClusterView(Pod pod)
         {
-            pod.ContainerImage = pod.ContainerImage.Substring(0, pod.ContainerImage.IndexOf(':'));
+            // If the container image is "image:tag", strip the ":tag", otherwise leave it alone
+            // not all images are tagged, so..
+            if(pod.ContainerImage.Contains(':'))
+                pod.ContainerImage = pod.ContainerImage.Substring(0, pod.ContainerImage.IndexOf(':'));
 
-            if (Pods.Any(x => x.Container == pod.Container))
+            if (Pods.Any(x => x.Name == pod.Name))
```
Comment (bradygmsft) [inline parrot/Hubs/Daemon.cs]: and another!?!?!? ugh. never let me code again.

Comment (sabbour) [reply parrot/Hubs/Daemon.cs]: At least it was consistent haha 😆

**Pull request merged**